examples/unixfs-file-cid/main.go

```diff
@@ -0,0 +1,231 @@
+	// to obtain a valid host ID.
+	priv, _, err := crypto.GenerateKeyPairWithReader(crypto.RSA, 2048, r)
+	if err != nil {
+		return nil, err
+	}
+
+	// Some basic libp2p options, see the go-libp2p docs for more details
+	opts := []libp2p.Option{
+		libp2p.ListenAddrStrings(fmt.Sprintf("/ip4/127.0.0.1/tcp/%d", listenPort)), // port we are listening on, limiting to a single interface and protocol for simplicity
+		libp2p.Identity(priv),
+	}
+
+	return libp2p.New(opts...)
+}
+
+func getHostAddress(h host.Host) string {
+	// Build host multiaddress
+	hostAddr, _ := multiaddr.NewMultiaddr(fmt.Sprintf("/p2p/%s", h.ID().String()))
+
+	// Now we can build a full multiaddress to reach this host
+	// by encapsulating both addresses:
+	addr := h.Addrs()[0]
+	return addr.Encapsulate(hostAddr).String()
+}
+
+// The CID of the file with the number 0 to 100k, built with the parameters:
+// CIDv1 links, a 256bit sha2-256 hash function, raw-leaves, a balanced layout, 256kiB chunks, and 174 max links per block
+const fileCid = "bafybeiecq2irw4fl5vunnxo6cegoutv4de63h7n27tekkjtak3jrvrzzhe"
+
+// createFile0to100k creates a file with the number 0 to 100k
+func createFile0to100k() ([]byte, error) {
+	b := strings.Builder{}
+	for i := 0; i <= 100000; i++ {
+		s := strconv.Itoa(i)
+		_, err := b.WriteString(s)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return []byte(b.String()), nil
+}
+
+func startDataServer(ctx context.Context, h host.Host) (cid.Cid, *bsserver.Server, error) {
+	fileBytes, err := createFile0to100k()
+	if err != nil {
+		return cid.Undef, nil, err
+	}
+	fileReader := bytes.NewReader(fileBytes)
+
+	ds := dsync.MutexWrap(datastore.NewMapDatastore())
+	bs := blockstore.NewBlockstore(ds)
+	bs = blockstore.NewIdStore(bs) // handle identity multihashes, these don't require doing any actual lookups
+
+	bsrv := blockservice.New(bs, offline.Exchange(bs))
+	dsrv := merkledag.NewDAGService(bsrv)
+
+	// Create a UnixFS graph from our file, parameters described here but can be visualized at https://dag.ipfs.tech/
+	ufsImportParams := uih.DagBuilderParams{
+		Maxlinks:  uih.DefaultLinksPerBlock, // Default max of 174 links per block
+		RawLeaves: true,                     // Leave the actual file bytes untouched instead of wrapping them in a dag-pb protobuf wrapper
+		CidBuilder: cid.V1Builder{ // Use CIDv1 for all links
+			Codec:    uint64(multicodec.DagPb),
+			MhType:   uint64(multicodec.Sha2_256), // Use SHA2-256 as the hash function
+			MhLength: -1,                          // Use the default hash length for the given hash function (in this case 256 bits)
+		},
+		Dagserv: dsrv,
+		NoCopy:  false,
+	}
+	ufsBuilder, err := ufsImportParams.New(chunker.NewSizeSplitter(fileReader, chunker.DefaultBlockSize)) // Split the file up into fixed sized 256KiB chunks
+	if err != nil {
+		return cid.Undef, nil, err
+	}
+	nd, err := balanced.Layout(ufsBuilder) // Arrange the graph with a balanced layout
+	if err != nil {
+		return cid.Undef, nil, err
+	}
```

Comment:
Wow... this is just a lot. I could simplify down by using some of the helpers that just use defaults. However:
1. I'd like the examples to use things closer to recommended usage (e.g. definitely raw leaves, and also CIDv1 links)
2. All this complexity seems like roughly the minimum to do non-default things and finding out how to change things to match the kubo CLI shouldn't be miserable for people

Hopefully some moving of packages into this repo + adding some nicer usability wrappers will help here. Although maybe just adding some docs and having a struct that makes more sense (e.g. `NoCopy` is really tied to how https://github.com/ipfs/go-filestore works rather than being user configurable)